core/src/main/java/org/mule/runtime/core/internal/util/MessagingExceptionResolver.java

```diff
@@ -148,13 +149,18 @@ private void propagateAlreadyLogged(MessagingException origin, MuleException res
         .findFirst();
   }
 
-  private List<Pair<Throwable, ErrorType>> collectErrors(Component obj, MessagingException me, ErrorTypeLocator locator) {
+  private List<Pair<Throwable, ErrorType>> collectUnsuppressedErrors(Component obj, MessagingException me,
+                                                                     ErrorTypeLocator locator) {
     List<Pair<Throwable, ErrorType>> errors = new ArrayList<>(4);
     final List<Throwable> exceptionsAsList = getExceptionsAsList(me);
     for (Throwable e : exceptionsAsList) {
-      ErrorType type = errorTypeFromException(obj, locator, e);
-      if (!isUnknownMuleError(type) && !isCriticalMuleError(type)) {
-        errors.add(new Pair<>(e, type));
+      if (e instanceof SuppressedMuleException) {
```

Comment:
handle this inside `getExceptionsAsList(me)`, so adding unwanted elements to the list is avoided from the start